src/main/java/com/jhuguet/sb_taskv1/app/services/impl/UserServiceImpl.java

```diff
@@ -117,40 +116,62 @@ private Tag countTagIterations(Order order) throws IdNotFound, NoTagInOrder {
 
     private Map<Integer, Integer> mapCurrentTags() {
         Map<Integer, Integer> currentTags = new HashMap<>();
-        tagRepository
-                .findAll()
-                .forEach(t -> currentTags.put(t.getId(), 0));
+        tagRepository.findAll().forEach(t -> currentTags.put(t.getId(), 0));
         return currentTags;
     }
 
     private Integer getMaxTagID(Map<Integer, Integer> tags) {
-        return tags
-                .entrySet()
-                .stream()
-                .max(Map.Entry.comparingByValue())
-                .get()
-                .getValue();
+        return tags.entrySet().stream().max(Map.Entry.comparingByValue()).get().getValue();
     }
 
     @Override
-    public User signIn(User user) throws MissingEntity, MissingRequiredFields {
+    public void signIn(User user) throws MissingEntity, MissingRequiredFields, UnqualifiedAuthority {
         if (!Objects.isNull(user)) {
             userEntityValidations(user);
-            userRepository.save(user);
+            userRepository.save(assembleUserToSave(user));
         } else {
             throw new MissingEntity();
         }
-        return null;
     }
 
-    private void userEntityValidations(User user) throws MissingRequiredFields {
-        if (user
-                .getPassword()
-                .equals("") || user
-                .getUsername()
-                .equals("")) {
+    private void userEntityValidations(User user) throws MissingRequiredFields, UnqualifiedAuthority {
+        if (user.getPassword().equals("") || user.getUsername().equals("")) {
             throw new MissingRequiredFields();
         }
+
+        if (user.getUsername().equalsIgnoreCase("administrator")) {
+            throw new UnqualifiedAuthority();
+        }
+    }
+
+    private User assembleUserToSave(User user) {
+        return new User(user.getUsername(), user.getEmail(), passwordEncoder.encode(user.getPassword()),
+                new HashSet<>(), "USER");
+    }
+
+    public void matchPasswords(String username, String password) throws WrongCredentials {
+        User user = userRepository.findByUsername(username);
+        if (!passwordEncoder.matches(password, user.getPassword()) && !user.getUsername().equalsIgnoreCase(
+                "administrator")) {
+            throw new WrongCredentials();
+        }
+    }
+
+    public void checkIdentity(String jwt, boolean requiresAdmin) throws NotAuthorized, IOException,
```

Comment:
What is the additional logic related to admin during checking identity? Could you explain?

Reply:
When performing a request, I'm validating whether that specific request requires an administrator permission to be able perform and resolve it. If the user that makes the request does not have and admin role, then an exception will pop up. 

But I guess that by the way the app is structured, the jwt and security configuration should take care of this.